server/src/main/java/org/elasticsearch/index/mapper/RuntimeField.java

```diff
@@ -58,49 +63,70 @@ default XContentBuilder toXContent(XContentBuilder builder, Params params) throw
 
     /**
      *  For runtime fields the {@link RuntimeField.Parser} returns directly the {@link MappedFieldType}.
-     *  Internally we still create a {@link RuntimeField.Builder} so we reuse the {@link FieldMapper.Parameter} infrastructure,
-     *  but {@link RuntimeField.Builder#init(FieldMapper)} and {@link RuntimeField.Builder#build(ContentPath)} are never called as
-     *  {@link RuntimeField.Parser#parse(String, Map, Mapper.TypeParser.ParserContext)} calls
-     *  {@link RuntimeField.Builder#parse(String, Mapper.TypeParser.ParserContext, Map)} and returns the corresponding
-     *  {@link MappedFieldType}.
      */
-    abstract class Builder extends FieldMapper.Builder {
-        final FieldMapper.Parameter<Map<String, String>> meta = FieldMapper.Parameter.metaParam();
+    abstract class Builder implements ToXContent {
+        final String name;
+        final Parameter<Map<String, String>> meta = Parameter.metaParam();
+
+        private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(RuntimeField.class);
 
         protected Builder(String name) {
-            super(name);
+            this.name = name;
         }
 
         public Map<String, String> meta() {
             return meta.getValue();
         }
 
-        @Override
-        protected List<FieldMapper.Parameter<?>> getParameters() {
+        protected List<Parameter<?>> getParameters() {
             return Collections.singletonList(meta);
         }
 
-        @Override
-        public FieldMapper.Builder init(FieldMapper initializer) {
-            throw new UnsupportedOperationException();
-        }
+        protected abstract RuntimeField createRuntimeField(Mapper.TypeParser.ParserContext parserContext);
 
         @Override
-        public final FieldMapper build(ContentPath context) {
-            throw new UnsupportedOperationException();
+        public final XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
I've been wondering why the builder is only responsible to print out its parameters. Could we include RuntimeField#toXContent in this so that we don't need to wrap it? If we do so maybe the toXContent within AbstractScriptFieldType could take a RUntimeField.Builder argument instead of ToXCOntent?

Reply:
let's see if we can do this as a follow-up, not important for merging